### DISCUSSION TITLE
Type Script Task

### DISCUSSION THREAD

**Discussion by saugatacr7** [akash-coded/mern#158] [Q&A]
1. Creating a Calculator
A. const operator = prompt('Enter operator ( either +, -, * or / ): ');

// take the operand input
const number1 = parseFloat(prompt('Enter first number: '));
const number2 = parseFloat(prompt('Enter second number: '));

let result;

// using if...else if... else
if (operator == '+') {
    result = number1 + number2;
}
else if (operator == '-') {
    result = number1 - number2;
}
else if (operator == '*') {
    result = number1 * number2;
}
else {
    result = number1 / number2;
}

// display the result
console.log(`${number1} ${operator} ${number2} = ${result}`);

2. Create a typescript array of numbers and create a variable to store the sum of all the numbers in the array. Compute the sum using a for loop and print the result to the console.
A. var arr=[7,15,29,7,8];
var sum=0;
for (let i=0; i< arr.length;i++){
    sum += arr[i];
}
   console.log("Sum is " + sum) 
3.  Create an array of any type and print the sum of all the numbers in the array to the console.
function sumArray(array) {
  const ourArray = [1, 4, 0, 9, -3];
  let sum = 0;
  for (let i = 0; i < ourArray.length; i += 1) {
    sum += ourArray[i];
  }
  
  return sum;
}
console.log(sumArray([1, 4, 0, 9, -3]));
5. Create an array of strings and print the strings that have a length of 5 to the console.

const xyz = ['bat', 'ball', 'mat', 'pad', 'gloves'];

console.log(xyz.length);
6. Create a boolean array and print the number of true values in the array to the console
var obj = [
   {
      isMarried: true
   },
   {
      isMarried: false
   },
   {
      isMarried: true
   },
   {
      isMarried: true
   },
   {
      isMarried: false
   }
]
function numberOfTrueValues(obj) {
   var counter = 0;
   for (var index = 0; index < obj.length; index++) {
      if (obj[index].isMarried === true) {
         counter++;
      }
   }

**Comment by Ruchiarth** [comment]
```ts

// Task 1 - Calculator app in TypeScript
//  enter the operation to perform hardcore value 
  let opn: string = "*";
  //  enter the first number as hardcore value  
  let val1: number = 5;
  // enter the second number as hardcore value  
  let val2: number = 6;
  
  // Perform the requested operation and log the result to the console
  switch (opn) {
    case "+":
      console.log(`${val1} + ${val2} = ${add(val1, val2)}`);
      break;
    case "-":
      console.log(`${val1} - ${val2} = ${subtract(val1, val2)}`);
      break;
    case "*":
      console.log(`${val1} * ${val2} = ${multiply(val1, val2)}`);
      break;
    case "/":
      console.log(`${val1} / ${val2} = ${divide(val1, val2)}`);
      break;
    default:
      console.log("Invalid operation");
      break;
  }
  
// addition
function add(n1: number, n2: number): number {
    return n1+n2;
  }
  
  // subtraction
  function subtract(n1: number,n2: number): number {
    return n1-n2;

  }
  
  // Define a function to perform multiplication
  function multiply(n1: number, n2: number): number {
    return n1 * n2;
  }
  
  // Define a function to perform division
  function divide(n1: number, n2: number): number {
    if (n2 === 0) {
      throw new Error("Can't divide by zero");
    }
    return n1 / n2;
  }

```

  
> # screenshot
> ![image](https://user-images.githubusercontent.com/126975575/233084514-0db0892d-7f0b-4962-b1cb-14ac158243d8.png)


``` ts
// Task2 : Create a typescript array of numbers and create a variable to store the sum of all the numbers in the array. 
// Compute the sum using a for loop and print the result to the console.

let arrnum: number[]=[10,20,30,40];

let total :number=0;
let cntr :number;

for(cntr=0;cntr<arrnum.length;cntr++)
{
    total = arrnum[cntr]+total;

}
console.log("Total is ",total);
```
# screenshot
![image](https://user-images.githubusercontent.com/126975575/233086865-de4a041e-0714-43a6-8ce4-faa28a1e58c3.png)

```ts
//Task 3 : Create an array of any type and print the sum of all the numbers 
//in the array to the console. 

let numar:any[]= ["Ruchi",10,"4.5",true,"Sharma",20,1.5];
let totalnum:number=0;

for(let cntr=0;cntr<numar.length;cntr++)
{


if(typeof(numar[cntr])=="number")
{
    totalnum = numar[cntr]+ totalnum; 
}

}
console.log(totalnum); 

```
# screenshot
![image](https://user-images.githubusercontent.com/126975575/233087482-53a624d9-5e7c-4478-88ca-e47b5dbfbfa7.png)

```ts
// Task5 Create an array of strings and print the strings that have a length of 5 to the console.

let strgarr:string[]=["Ruchi","Rashmi","Mahavir","Excel","Level"];


for(let cntr=0;cntr<strgarr.length;cntr++)
{ 
if(strgarr[cntr].length==5){

    console.log(strgarr[cntr]);
}
}

```
# screenshot
![image](https://user-images.githubusercontent.com/126975575/233087891-0779aef2-5ff7-46a7-9200-b91b2c9c121d.png)
```ts

//Task6 : Create a boolean array and print the number of true values in the array to the console. 
let barr:boolean[] =[true,false,true,true, false, true];

let pcntr:number=0;

for(let cntr=0;cntr<barr.length;cntr++)
{
    if(barr[cntr])
    {
        pcntr=pcntr+1;

    }

}
console.log("The number of true values in Boolean Array is " + pcntr);

```
# screenshot
![image](https://user-images.githubusercontent.com/126975575/233088460-eea94ee2-b9af-4e44-9bbe-9a6cc56d4237.png)

``` ts
// Task7 :Create five variables of each of the primitive types and print their values to the console.
 
let pNumber: number = 42;
let pString: string = "These are primitive types ";
let pBoolean: boolean = true;
let pNull: null = null;
let pUndefined: undefined = undefined;

console.log(pNumber); 
console.log(pString); 
console.log(pBoolean);
console.log(pNull); 
console.log(pUndefined); 


```
# screenshot
![image](https://user-images.githubusercontent.com/126975575/233088946-5a449e1b-f60a-411a-85e6-ef5312ae5017.png)
